### ISSUE TITLE
Failed Handshake ?

### ISSUE THREAD

**Issue by Chaiavi** [AsyncHttpClient/async-http-client#1663]
While trying to fetch a page from [PcGamer](https://www.pcgamer.com/uk/best-pc-games/)

The client failed with this exception
`java.util.concurrent.ExecutionException: java.net.ConnectException: Received fatal alert: handshake_failure

	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357)
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1895)
	at org.asynchttpclient.netty.NettyResponseFuture.get(NettyResponseFuture.java:201)`

Caused by
`Caused by: javax.net.ssl.SSLException: Received fatal alert: handshake_failure
	at sun.security.ssl.Alerts.getSSLException(Alerts.java:208)
	at sun.security.ssl.SSLEngineImpl.fatal(SSLEngineImpl.java:1666)
	at sun.security.ssl.SSLEngineImpl.fatal(SSLEngineImpl.java:1634)
`

Etc...

My scenario is:
```
DefaultAsyncHttpClientConfig.Builder config = config()
        .setCompressionEnforced(true)
        .setDisableHttpsEndpointIdentificationAlgorithm(true)
        .setFollowRedirect(true)
        .setUserAgent("gdsd")
        .setUseInsecureTrustManager(true);
    asyncHttpClient = asyncHttpClient(config);
```

```
Future<Response> future = asyncHttpClient.prepareGet(urlStr).execute();
    Response response = future.get();
```

What is wrong ?

**Comment by slandelle** [comment]
Works for me with master branch.

```
import org.asynchttpclient.AsyncHttpClient;
import org.asynchttpclient.Response;

import java.util.concurrent.Future;

import static org.asynchttpclient.Dsl.*;

public class Main {

  public static void main(String[] args) throws Exception {
    try (AsyncHttpClient asyncHttpClient = asyncHttpClient(
      config()
        .setCompressionEnforced(true)
        .setDisableHttpsEndpointIdentificationAlgorithm(true)
        .setFollowRedirect(true)
        .setUserAgent("gdsd")
        .setUseInsecureTrustManager(true)
    )) {
      Future<Response> future = asyncHttpClient.prepareGet("https://www.pcgamer.com/uk/best-pc-games/").execute();
      Response response = future.get();
      System.out.println(response);
    }
  }
}
```

Which version do you use?

**Comment by slandelle** [comment]
FYI, I've just released 2.10.2 .

**Comment by Chaiavi** [comment]
I used 2.10.1

Will try now with newest version

**Comment by Chaiavi** [comment]
It is not in Maven repo yet

**Comment by Chaiavi** [comment]
I am using now latest (2.10.2) and it still fails

**Comment by Chaiavi** [comment]
I am using Java8

Maybe you used a later version?

**Comment by slandelle** [comment]
```
$ java -version
openjdk version "1.8.0_202"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_202-b08)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.202-b08, mixed mode)
```

If you're running an old JDK8, you could be missing modern ciphers.

**Comment by Chaiavi** [comment]
I don't really understand it, but when I tried using UniRest fetcher it worked although AHC failed.

But when I updated the Java version to v8-Latest it worked on all fetchers - so thank you (It might be worth checking the trick UniRest fetcher uses which allowed it to work with old Java versions, but that is just an uneducated observation).

Thank you for everything, it works now like a charm!


Avi.
